src/win/fs.c

```diff
@@ -606,6 +606,14 @@ void fs__open(uv_fs_t* req) {
     goto einval;
   }
 
+  /* Copy existing attributes to prevent an "Access is denied" error for hidden
+   * and system files. */
+  DWORD existing_attributes = GetFileAttributesW(req->file.pathw);
+  if (existing_attributes != INVALID_FILE_ATTRIBUTES) {
+    attributes |= (existing_attributes
+      & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM));
+  }
```

Comment:
This seems like it's vulnerable to a TOCTOU bug between the GetFileAttributes() and CreateFile() calls.

Possibly it's harmless but I don't do enough Windows programming to say for sure. @libuv/windows PTAL.

Reply:
@bnoordhuis How about this: https://github.com/sryze/libuv/commit/3b176454f1cfded4658f378ec4708b4bc2a1ba19
Is it better? It only checks for *_HIDDEN flag though. I'm not sure if it's a good idea to have 3 more CreateFile calls instead of 1 (for 2 more possible combinations with *_SYSTEM attribute).

Reply:
Or maybe it's better to let the caller pass a "hidden" flag from outside, like it's done for `mode` and `flags`?

Reply:
Retrying on "access denied" seems a good way forward.

Apropos FILE_ATTRIBUTE_SYSTEM, I don't know if that's something libuv ought to be setting. Presumably the attribute is intended as a "hands off" signal.

Reply:
OK, I've updated the code to use the 2nd approach